### DISCUSSION TITLE
Could I get a video tutorial on building for windows?

### DISCUSSION THREAD

**Discussion by ghost** [valhalla/valhalla#3862] [Q&A]
I'm stuck on instruction 2 for installing to windows, I don't really know what i'm doing and I work better visually you see and have only quite basic programming experience, i'd appreciate having a video to follow along with (doesn't have to be lord of the rings quality either) so I can just get it setup. Would you be willing to provide that or point me to one? All I can find is how to setup Assassins Creed Valhalla on Youtube which is no use to me... Appreciate any help and thanks for taking the time to read this :)

Also apologies if this is in the wrong category, I didn't think this counted as an idea for a feature but more of a question so I stuck in Q&A

**Comment by kevinkreiser** [comment]
i could not resist:
![image](https://user-images.githubusercontent.com/697548/206550688-8061b99b-1122-4605-820a-fb9f065d8b8c.png)



seriously though, it took me a long time to figure out how to compile valhalla on windows and i havent done it in a very long time. we only have a couple people who regularly use windows with valhalla (that contribute enough that i know they exist) so i think if it has to be windows you might be waiting a long time.

another option, would be to try on our best supported platform (linux). these days it is very easy to run linux stuff from inside windows by doing so with docker. in fact we provide docker containers and gis-ops provides even nicer containers that you can pick up and just run with. you'll still be in windows but linux will run on top of it inside a docker container with valhalla inside of that. the performance should be just as good as native. in truth, just about all production environments these days (think of all the big silicon valley companies) are running their services on docker anyways. this is because it brings a lot of reproducability and operational cleanliness but also because so many fools insist on using macs for their development... but i digress. but the bonus here is that you will find tons of videos on people showing you how to use docker to run someone else's software.

**Comment by nilsnolde** [comment]
Yeah I feel like docker or WSL2 should be our recommendation in the README to install on Windows. It's not supposed to be a dev guide, more how-to-run guide and it's really involved. First time I had to do it I was probably not far away from your level and it took me aaaages. In fact I got a new PC yesterday and even before dual-booting Linux I tried installing Valhalla and failed! And I wrote the Windows guide and maintain much of it. Life's too short to waste your time to get the native installation work, unless you really need it for some custom stuff:)

**Reply by ghost** [reply]
I tried installing openroute service through docker but docker wouldn't run for me (not even the old toolbox version), it's cos I don't have bios visualisation enabled on my home pc (and cannot enable it) so I doubt valhalla and docker will work if docker is the problem. The project i'm working on is basically creating a route along a lot of roads (traveling salesman problem sort of), retrieving the coordinates of each point and converting them to lines i.e point A to point B is one line, Point B to point C is the second line and so on, I have scratch code that does the conversion and it works well, I will then store these lines iin an xml sorted into arrays and draw them in a garmin watch app, 

The main problem is I just need a way to create a really long route that snaps to roads (I tried with the free software garmin basecamp but routes would often switch to crow flies if you made the next point in the route too far from the previous one) but many softwares with better snappage cost money/subscription (which I a lowley student during a cost of living crisis canny afford) to do anything longer than a certain amount of stops. Then I found Qgis (which i'd used a little for the maps) had valhalla and ORS plugins, unfortunately there is a limit on distance for the routes :( so I wanted to see if hosting the route generation myself using ors or valhalla open source on my pc would give me unlimited route lengths (not that i'm planning to route the whole world) whereupon I could then make the plugins call locally using some sort of request (basically whatever way your system allows me to e.g an api key).

i might look into WSL2, as I haven't heard of that so maybe i'll find luck there?

O'course if you have any recommendations for free routing software easily installable with unlimited stops and auto following roads i'd be content with that :|

**Reply by nilsnolde** [reply]
> I don't have bios visualisation enabled on my home pc (and cannot enable it) 

Yeah that sucks, then docker is not an option. And WSL2 needs that pretty sure too.. So either you dual-boot Linux, which I can highly recommend for the long run if you wanna start with some development. Honestly, I started the same as you many years ago. If you don't have the pressure, try to learn the easier way and install Linux next to Windows on the same computer. Not only will it feel much better to ditch proprietary horse shit like Windows (even Microsoft realized that and gave us WSL2 and there's no LS2 ;)), you'll realize that your computer can be much faster. Far less stuff running in the background by default.

I started my routing journey working for ORS in Heidelberg and I only tried installing it once on Windows before I decided to switch to Linux full-time. And I know very well how painful Valhalla is on Windows. So do it man, now is the time! :laughing:
